### PR TITLE
docs: Add graphiql create-react-app example.

### PR DESCRIPTION
**Pull request by daemon1024**
Ref: #1502 

- [X] create react app example
- [x] convert to typescript

### CODE REVIEW

**Comment by daemon1024** [comment]
> this looks good so far! lets make it use typescript and that should be it.

Ok will convert to typescript.

>Lets make this PR just the CRA example. like I mentioned, we want to do one PR at a time, instead of all the examples at once.

Cool.

**Comment by daemon1024** [comment]
@acao I hope it's complete now. Kindly review it. If you approve then can move on to next example :)

File: examples/graphiql-create-react-app/public/index.html [hunk 1]
```diff
@@ -0,0 +1,31 @@
+<!DOCTYPE html>
```
Comment (acao) [inline examples/graphiql-create-react-app/public/index.html:1]: did create react app generate this file?

File: examples/graphiql-create-react-app/package.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
+{
+  "name": "example-graphiql-create-react-app",
+  "version": "0.1.0",
+  "private": true,
+  "dependencies": {
+    "@types/react": "^16.9.34",
+    "@types/react-dom": "^16.9.6",
+    "graphiql": "file:../../packages/graphiql",
```
Comment (acao) [inline examples/graphiql-create-react-app/package.json:8]: this should be `file:packages/graphiql` now, my bad

File: examples/graphiql-create-react-app/package.json [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    "react-scripts": "3.4.1"
+  },
+  "scripts": {
+    "start": "react-scripts start",
+    "build": "react-scripts build"
+  },
+  "eslintConfig": {
+    "extends": "react-app"
+  },
+  "browserslist": {
```
Comment (acao) [inline examples/graphiql-create-react-app/package.json:21]: if this was not added by CRA we should remove it. only CRA-related changes except adding GraphiQL component. I haven't used CRA in a few years really so I'm not sure

File: examples/graphiql-create-react-app/package.json (hunk 2 above)
Comment (acao) [reply examples/graphiql-create-react-app/package.json:8]: to confirm this resolves correctly in yarn workspaces, run `yarn workspaces info` and make sure there aren't any `mismatchedDependencies`

File: examples/graphiql-create-react-app/package.json (hunk 3 above)
Comment (daemon1024) [reply examples/graphiql-create-react-app/package.json:21]: Yup it was added by CRA.

File: examples/graphiql-create-react-app/public/index.html (hunk 1 above)
Comment (daemon1024) [reply examples/graphiql-create-react-app/public/index.html:1]: Yes it was generated by CRA, I modified the meta info a bit to suit the context.

File: examples/graphiql-create-react-app/package.json (hunk 2 above)
Comment (daemon1024) [reply examples/graphiql-create-react-app/package.json:8]: > this should be `file:packages/graphiql` now, my bad

That doesn't seem to work. 
It throws a error `Package "graphiql" refers to a non-existing file . . .`. 
It points to the `packages/graphiql` inside the `examples/cra`

Comment (daemon1024) [reply examples/graphiql-create-react-app/package.json:8]: There aren't any `mismatchedDepedencies`

Comment (acao) [reply examples/graphiql-create-react-app/package.json:8]: ok, I think I had something backwards then, my bad, carry on!

**Comment by acao** [comment]
great work, thank you @daemon1024

**Pull request merged**